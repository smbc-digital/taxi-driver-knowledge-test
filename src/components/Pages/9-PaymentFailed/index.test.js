import React from 'react'
import Enzyme, { mount } from 'enzyme'
import queryString from 'query-string'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import PaymentFailed from './index'

Enzyme.configure({ adapter: new Adapter() })

describe('PaymentFailed', () => {

    it('should parse payment url query string', () => {
        // Arrange
        let parseMock = jest.fn()
        parseMock.mockReturnValue({
            paymenturl: 'test'
        })
        queryString.parse = parseMock

        mount(<PaymentFailed location={{ search: '?paymenturl=test' }} />)

        // Assert
        expect(parseMock).toHaveBeenCalled()
    })

    describe('snapshot', () => {

        it('should render correctly', () => {
            // Arrange
            const tree = renderer
                .create(<PaymentFailed location={{ search: '?paymenturl=test' }} />)
                .toJSON()

            // Assert
            expect(tree).toMatchSnapshot()
        })
    })
})