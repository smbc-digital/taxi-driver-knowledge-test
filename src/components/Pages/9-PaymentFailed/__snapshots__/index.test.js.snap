// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentFailed snapshot should render correctly 1`] = `
<form>
  <div
    className="ineligible-warning"
  >
    <div
      className="header-container"
    >
      <div>
        <i
          aria-hidden="true"
          className="fa fa-exclamation"
        />
      </div>
      <h1>
         We're sorry, the payment was declined
      </h1>
    </div>
    <p
      className="no-margin-bottom"
    >
      Please try again if you want to continue with the order.
    </p>
    <a
      className="button-primary"
      href="test"
    >
      Continue with the order
    </a>
    <a
      className="button-secondary anchor-as-block"
      href="https://www.stockport.gov.uk/"
    >
      Go to the homepage
    </a>
  </div>
</form>
`;
